v11y/api/src/db/mod.rs

```diff
@@ -322,6 +356,49 @@ impl Db {
             }
         })
     }
+
+    pub async fn get_related<'s>(
+        &'s self,
+        id: &'s str,
+        origin: Option<String>,
+    ) -> impl Stream<Item = (String, String)> + 's {
+        let query = if let Some(origin) = origin {
+            sqlx::query(
+                r#"
+                select
+                    origin, related
+                from
+                    related
+                where
+                    vulnerability_id = $1 and origin = $2
+                "#,
+            )
+            .bind(id)
+            .bind(origin)
+        } else {
+            sqlx::query(
+                r#"
+                select
+                    origin, related
+                from
+                    related
+                where
+                    vulnerability_id = $1
+                order by
+                    origin
+                "#,
+            )
+            .bind(id)
+        };
+
+        query.fetch(&self.pool).filter_map(|row| async move {
+            if let Ok(row) = row {
+                Some((row.get::<String, _>("origin"), row.get::<String, _>("related")))
+            } else {
+                None
+            }
+        })
```

Comment:
I'm being pedantic, but the `else { None }` bothers me. I'm surprised clippy is ok with it. :smile: 

I think this is equivalent...
```suggestion
        query
            .fetch(&self.pool)
            .filter_map(|row| async { row.ok().map(|v| (v.get("origin"), v.get("related"))) })
```

Reply:
Thanks, it felt weird but I'd gone dumb. Will improve.